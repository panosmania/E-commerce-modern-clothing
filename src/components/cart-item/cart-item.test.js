import { shallow } from "enzyme";
import React from "react";
import toJson from "enzyme-to-json";

import CartItem from "./cart-item.component";

describe("CartItem", () => {
  const mockProps = {
    imageUrl: "www.testImage.com",
    price: 10,
    name: "hats",
    quantity: 2,
  };

  const wrapper = shallow(<CartItem item={mockProps} />);

  it("should render CartItem component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render imageUrl: www.testImage.com ", () => {
    //wrapper.find(".price");
    expect(mockProps.imageUrl).toEqual("www.testImage.com");
  });

  it("should render price: toBe 10 ", () => {
    //wrapper.find(".price");
    expect(mockProps.price).toBe(10);
  });

  it("should NOT render name: hats ", () => {
    //wrapper.find(".price");
    expect(mockProps.name).not.toMatch("Bob");
  });
});
