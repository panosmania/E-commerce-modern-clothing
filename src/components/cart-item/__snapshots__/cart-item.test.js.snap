// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem should render CartItem component 1`] = `
<div
  className="cart-item"
>
  <img
    alt="item"
    src="www.testImage.com"
  />
  <div
    className="item-details"
  >
    <span
      className="name"
    >
      hats
    </span>
    <span
      className="price"
    >
      2
       x $
      10
    </span>
  </div>
</div>
`;
